toolkits/asana/evals/eval_workspaces.py

```diff
@@ -0,0 +1,173 @@
+                            "arguments": '{"limit":2}',
+                        },
+                    }
+                ],
+            },
+            {
+                "role": "tool",
+                "content": json.dumps({
+                    "count": 2,
+                    "workspaces": [
+                        {
+                            "gid": "1234567890",
+                            "name": "Workspace Hello",
+                        },
+                        {
+                            "gid": "1234567891",
+                            "name": "Workspace World",
+                        },
+                    ],
+                }),
+                "tool_call_id": "call_1",
+                "name": "Asana_ListWorkspaces",
+            },
+            {
+                "role": "assistant",
+                "content": "Here are two workspaces in Asana:\n\n1. Workspace Hello\n2. Workspace World",
+            },
+        ],
+    )
+
+    suite.add_case(
+        name="List workspaces with pagination changing the limit",
+        user_message="Show me the next 5 workspaces.",
+        expected_tool_calls=[
+            ExpectedToolCall(
+                func=list_workspaces,
+                args={
+                    "limit": 5,
+                    "offset": 2,
+                },
+            ),
+        ],
+        rubric=rubric,
+        critics=[
+            BinaryCritic(critic_field="limit", weight=0.5),
+            BinaryCritic(critic_field="offset", weight=0.5),
+        ],
+        additional_messages=[
+            {"role": "user", "content": "Show me 5 workspaces in Asana."},
+            {
+                "role": "assistant",
+                "content": "",
+                "tool_calls": [
+                    {
+                        "id": "call_1",
+                        "type": "function",
+                        "function": {
+                            "name": "Asana_ListWorkspaces",
+                            "arguments": '{"limit":5}',
+                        },
+                    }
+                ],
+            },
+            {
+                "role": "tool",
+                "content": json.dumps({
+                    "count": 2,
+                    "workspaces": [
+                        {
+                            "gid": "1234567890",
+                            "name": "Workspace Hello",
+                        },
+                        {
+                            "gid": "1234567891",
+                            "name": "Workspace World",
+                        },
+                    ],
+                }),
+                "tool_call_id": "call_1",
+                "name": "Asana_ListWorkspaces",
+            },
+            {
+                "role": "assistant",
+                "content": "Here are five workspaces in Asana:\n\n1. Workspace Hello\n2. Workspace World\n3. Workspace Hello\n4. Workspace World\n5. Workspace Hello",
+            },
+        ],
```

Comment:
this needs to be updated. The tool returns 2 workspaces, but the message says there are five.